app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/DefaultTaskRepository.kt

```diff
@@ -136,15 +143,15 @@ class DefaultTaskRepository(
      * Also, in a real app, these operations could be scheduled using WorkManager.
      */
     private suspend fun loadTasksFromNetwork() {
-        withContext(coroutineDispatcher) {
+        withContext(expensiveWorkDispatcher) {
             val remoteTasks = tasksNetworkDataSource.loadTasks()
             taskDao.deleteAll()
             taskDao.upsertAll(remoteTasks.toLocal())
         }
     }
 
-    private suspend fun saveTasksToNetwork() {
-        withContext(coroutineDispatcher) {
+    private fun saveTasksToNetwork() {
+        fireAndForgetScope.launch {
```

Comment:
As discussing in Chat... what about errors here?

Reply:
Let's add a comment and that should be enough

Reply:
Done